# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/GtkCompositorWidget.cpp
# Commit: c1d946659c92
# Full Hash: c1d946659c921aaaf3d329e4af69c4b5b866b9e3
# Author: stransky <stransky@redhat.com>
# Date: 2025-02-19 21:53:46
# Regressor Bug: 1938033
# File Overlap Count: 1
# Description:
#   Bug 1938033 [Wayland] Implement NativeLayerWayland by WaylandSurface r=lsalzman,emilio
#   
#   Make NativeLayerRootWayland to use WaylandSurface owned by nsWindow widget and create WaylandSurface for every NativeLayerWayland.
#   WaylandSurface implementation is shared by native compositing code, SWWR and WR rendering so we don't duplicate the implementation here.
#   
# ==============================================================================

diff -r 6064ee48431c -r c1d946659c92 widget/gtk/GtkCompositorWidget.cpp
--- a/widget/gtk/GtkCompositorWidget.cpp	Wed Feb 19 13:54:16 2025 +0000
+++ b/widget/gtk/GtkCompositorWidget.cpp	Wed Feb 19 13:54:17 2025 +0000
@@ -55,6 +55,11 @@
 GtkCompositorWidget::~GtkCompositorWidget() {
   LOG("GtkCompositorWidget::~GtkCompositorWidget [%p]\n", (void*)mWidget.get());
   CleanupResources();
+#ifdef MOZ_WAYLAND
+  if (mNativeLayerRoot) {
+    mNativeLayerRoot->Shutdown();
+  }
+#endif
   RefPtr<nsIWidget> widget = mWidget.forget();
   NS_ReleaseOnMainThread("GtkCompositorWidget::mWidget", widget.forget());
 }